internal/hierarchyconfig/validator_test.go

```diff
@@ -17,6 +17,101 @@ import (
 	"sigs.k8s.io/hierarchical-namespaces/internal/objects"
 )
 
+func TestManagedMetaPermissions(t *testing.T) {
+	f := foresttest.Create("-") // a
+	h := &Validator{Forest: f}
+	l := zap.New()
+
+	tests := []struct {
+		name        string
+		labels      []api.MetaKVP
+		annotations []api.MetaKVP
+		allowed     bool
+	}{
+		{name: "managed label", labels: []api.MetaKVP{{Key: "label.com/team"}}, allowed: true},
+		{name: "unmanaged label", labels: []api.MetaKVP{{Key: "kubernetes.io/metadata.name"}}},
+		{name: "managed annotation", annotations: []api.MetaKVP{{Key: "annot.com/log-index"}}, allowed: true},
+		{name: "unmanaged annotation", annotations: []api.MetaKVP{{Key: "openshift.io/sa.scc.uid-range"}}},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			// Setup
+			g := NewWithT(t)
+			// In this test we only allow labels/annotations with the defined prefixes
+			err := config.SetManagedMeta([]string{"label\\.com/.*"}, []string{"annot\\.com/.*"})
+			g.Expect(err).ToNot(HaveOccurred())
+
+			hc := &api.HierarchyConfiguration{Spec: api.HierarchyConfigurationSpec{}}
+			hc.ObjectMeta.Name = api.Singleton
+			hc.ObjectMeta.Namespace = "a"
+			hc.Spec.Labels = tc.labels
+			hc.Spec.Annotations = tc.annotations
+			req := &request{hc: hc}
+
+			got := h.handle(context.Background(), l, req)
+
+			logResult(t, got.AdmissionResponse.Result)
+			g.Expect(got.AdmissionResponse.Allowed).Should(Equal(tc.allowed))
+		})
+	}
+}
+
+func TestManagedMetaSyntax(t *testing.T) {
```

Comment:
These two tests (...Permissions and ...Syntax) are basically identical. I'd just combine them myself. The structure of the test is basically the same and they're testing the same function.